Update from upstream branch

crayolakat opened the pull request.

Commit:
add phone to ngpvan apply canvass result (#705)

* first draft

* fixed error message

* switched from `input_type_id` to `contact_type_id`

* added tests

* linting

AndrewRook committed:
Added tests for SendMail notification connector (#703)

cmc333333 committed:
Fix error in VAN Target Export method. (#704)

* Fix error VAN Target Export method.

It looks like this method assumes VAN return a *list* of exported files
rather than a single target, but VAN currently only returns individual
targets: https://docs.ngpvan.com/reference/targetexportjobsexportjobid

They might've changed the API since this was last updated? In any event,
these tweaks return correct data in my recent testing!

SorenSpicknall committed:
Updated config.yml

SorenSpicknall committed:
Updated config.yml

SorenSpicknall committed:
HOTFIX Sendmail pep8 Compliance (#711)

* HOTFIX sendmail pep8 compliance

SorenSpicknall committed:
Update docker image version referenced in testing (#712)

shaunagm committed:
Update Mobilize connector URI, fix display of errors, fix auth issue (#708)

* Raises readable errors when unsuccessful status codes returned

* Remove extra space for linting purposes

* Switch URI to working one

* Add trailing slash to mobilize uri

* Fix auth issue in request_paginate

shaunagm committed:
Update setup.py for new release: 0.20.0 (#713)

neverett committed:
Bugfix: remove leading and trailing whitespace from schema and table name (#715)

cmc333333 committed:
More flexible importing in parsons/__init__.py (#702)

This implements the try-imports method @schuyler1d
[described](https://github.com/move-coop/parsons/pull/632#issuecomment-1065494121)
in a sustainable way, thereby allowing
```
from parsons import ...
```
for users without all of the dependencies.

SorenSpicknall committed:
Fix Redshift column alteration bug (#709)

* Fix Redshift column alteration bug that was resulting in too-wide varchar column requests

shaunagm committed:
Fix small bug with postal addresses in add_person, plus under-documented tags param (#716)

AndrewRook committed:
Sendmail connector is now an abstract base class (#717)

SorenSpicknall committed:
Bugfix alter_varchar_columns dependency issue (#720)

SorenSpicknall committed:
HOTFIX Sendmail tests linting (#721)

Jason94 committed:
Convert None to "" in convert_columns_to_str (#728)

Co-authored-by: Jason Walker <jason@indivisible.org>

benstroud committed:
TargetSmart 2022: SmartMatch (#674)

* TargetSmart 2022: SmartMatch

Adds support for performing TargetSmart SmartMatch list matching workflow executions. SmartMatch matches contact list records to Voterbase, TargetSmart's national database of voting age individuals.

In addition, code review of the previously implemented TargetSmart routines has been done to modernize for 2022, including doc updates.

shaunagm committed:
Add upsert_person method to ActionNetwork (#734)

* Add upsert_person method to ActionNetwork, add deprecation warnings to add_person and update_person

SorenSpicknall committed:
Enhancement: Allow Subfolders for Temp S3 Bucket in Redshift Operations (#738)

* Allow temporary buckets with subfolders in Redshift operations

sjwmoveon committed:
Don't auto-grab session token if it's not passed in (#740)

* Don't auto-grab session token if it's not passed in

It turns out that trying to grab a session token from the environment variable can break things when used in concert with Zappa asynchronous processing - the access key and secret key passed in may not match the temporary token generated for the asynchronous process. Thus, this change requires the token to be passed in, even if it's set as an environment variable. 

As described [here](https://github.com/zappa/Zappa#asynchronous-task-execution), with asynchronous task execution, Zappa spins up a completely separate Lambda instance. The session token is generated as part of that process, and (as far as I can tell) is based on the execution role of the Lambda function.
So when the asynchronous task is running, it has the `aws_session_token` set even though there wasn't one originally. I'm not sure why it isn't just "None".
In any case, the token doesn't match the access key ID and secret access key that I was being passed in, resulting in an InvalidTokenError: `An error occurred (InvalidToken) when calling the PutObject operation: The provided token is malformed or otherwise invalid`. It's not that the token is actually invalid, but it doesn't match the credentials being passed in.

I'd like to be able to force aws_session_token to None even if the environment variable is set.

* Add optional argument to ignore env token

* Spacing

Jason94 committed:
Update Mobilize connector (#733)

* Fix MobilizeAmerica.get_events_organization

* Update documentation

* Add promoted organizations endpoint

* Fix bug with max_timeslots = 0

* Fix get_people for multiple orgs

* Remove fake news attendance options

* Flake8 fixes

* PR review comments

* Add tests to timeslot logic

* Fix docstring label

* Linter fixes

Co-authored-by: Jason Walker <jason@indivisible.org>

Jason94 committed:
Default to utf-8 encoding writing to RS S3 (#732)

* Default to utf-8 encoding writing to RS S3

* Add csv_encoding parameter

Co-authored-by: Jason Walker <jason@indivisible.org>

shaunagm committed:
Update psycopg2-binary from 2.8.5 to 2.9.3 (#731)

SorenSpicknall committed:
Update version number for 0.21.0 release (#743)

crayolakat committed:
Merge branch 'main' into moveon_update

Commit:
Update cache keys for CircleCI configuration (#752)

sjwmoveon reviewed (changes requested):
Sorry, this is absolutely my fault - but I can't approve this until https://github.com/move-coop/parsons/pull/745 is merged as otherwise it will break fundraising-parsons-scripts.

alxmrs committed:
Typo fixes for ETL best practices guide. (#747)

* Typo fix for ETL best practices guide.

* Changed variable name to fix code example.

* Attempt to fix minor rst rendering error.

ethyoo committed:
Fix parsing of code blocks for Mobilize America (#749)

AndrewRook committed:
Simple docs for the SMTP connector (#735)

* added simple docs for STMP

* fixed formatting

sjwmoveon committed:
Add use_env_token option to S3 object (#745)

Previously this was only settable in AWSConnection, but you can't pass an AWSConnection to the S3 object.

sjwmoveon commented:
The pull request I mentioned has been merged - can you add that change as well?

crayolakat committed:
Merge branch 'main' into moveon_update

crayolakat commented:
@sjwmoveon Done!
